vignettes/tidyverse_translation.Rmd

```diff
@@ -21,16 +21,25 @@ knitr::opts_chunk$set(
 
 pkgs <- c(
   "dplyr",
-  "datawizard",
   "tidyr"
 )
+all_deps_available <- all(vapply(pkgs, requireNamespace, quietly = TRUE, FUN.VALUE = logical(1L)))
 
-# since we explicitely put eval = TRUE for some chunks, we can't rely on
-# knitr::opts_chunk$set(eval = FALSE) at the beginning of the script. So we make
-# a logical that is FALSE only if deps are not installed (cf easystats/easystats#317)
-evaluate_chunk <- TRUE
+if (all_deps_available) {
+  library(datawizard)
+  library(dplyr)
+  library(tidyr)
+}
+
+can_vignette_be_evaluated <- all_deps_available && getRversion() >= "4.1.0"
 
-if (!all(vapply(pkgs, requireNamespace, quietly = TRUE, FUN.VALUE = logical(1L))) || getRversion() < "4.1.0") {
+# Since we explicitly put `eval = TRUE` for some chunks, we can't rely on
+# `knitr::opts_chunk$set(eval = FALSE)` at the beginning of the script.
+# Therefore, we introduce a logical that is `FALSE` only if all suggested
+# dependencies are not installed (cf easystats/easystats#317)
+if (can_vignette_be_evaluated) {
+  evaluate_chunk <- TRUE
+} else {
```

Comment:
I don't understand this. Why not put `evaluate_chunk <- all_deps_available && getRversion() >= "4.1.0"` and remove this `if` condition?

Reply:
Good point. No need for an extra level of indirection.